lib/transaction.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'tracker'
+require_relative 'statement_formatter'
+require 'date'
+
+class Transaction
+  attr_reader :date, :amount
+
+  def initialize(date, amount)
+    check_date_if_valid(date)
+    check_amount_if_valid(amount)
+    @date = date
+    @amount = amount
+  end
+
+  private
+
+  def check_date_if_valid(date)
```

Comment:
Could you let the Ruby date parser do the hard work here by just passing it whatever the user inputs and rejecting it if an exception is thrown?